### PR TITLE
remove use of deprecated function readdir_r

### PR DESCRIPTION
**Pull request by evsh**
Ports 140b8ace onto RC_1_1 branch

Compilation with GCC 8 fails without such change.

### CODE REVIEW

File: src/file.cpp [hunk 1]
```diff
@@ -1185,7 +1185,6 @@ namespace libtorrent
 		}
 #else
 
-		memset(&m_dirent, 0, sizeof(dirent));
 		m_name[0] = 0;
```
Comment (arvidn) [inline src/file.cpp]: I notice you left this line in (`m_name[0] = 0;`) while it was removed in master. is it necessary in `RC_1_1`?

Comment (evsh) [reply src/file.cpp]: Overlooked, sorry. Of course, it is a mistake to assign to a empty `std::string`.

Comment (evsh) [reply src/file.cpp]: Assignment removed.

**Comment by evsh** [comment]
thanks!

**Pull request merged**